src/mesh/exodusII_io_helper.C

```diff
@@ -1343,6 +1343,46 @@ void ExodusII_IO_Helper::write_elements(const MeshBase & mesh, bool use_disconti
   elem_num_map.resize(n_active_elem);
   std::vector<int>::iterator curr_elem_map_end = elem_num_map.begin();
 
+  exII::ex_block_params blocks = {};
+  blocks.elem_blk_id = new int[num_elem_blk];
+  blocks.elem_type = new char*[num_elem_blk];
+  blocks.num_elem_this_blk = new int[num_elem_blk];
+  blocks.num_nodes_per_elem = new int[num_elem_blk];
+  blocks.num_edges_per_elem = new int[num_elem_blk];
+  blocks.num_faces_per_elem = new int[num_elem_blk];
+  blocks.num_attr_elem = new int[num_elem_blk];
```

Comment:
All the memory allocated for the blocks struct to use seems to be at best leaked, or at worst, incorrectly free'd by the Exodus API? Maybe just create some local `vector<int>`s here and have the `blocks` struct use that storage.